openpgp/keys.go

```diff
@@ -115,7 +115,7 @@ func (e *Entity) encryptionKey(now time.Time) (Key, bool) {
 	// assume that the primary key is ok. Or, if the primary key is
 	// marked as ok to encrypt to, then we can obviously use it.
 	i := e.primaryIdentity()
-	if !i.SelfSignature.FlagsValid || i.SelfSignature.FlagEncryptCommunications &&
+	if (!i.SelfSignature.FlagsValid || i.SelfSignature.FlagEncryptCommunications) &&
```

Comment:
Yeah this is a bad bug in OpenPGP, meaning encryptions can be send to expired keys.  Cc: @agl